options.go

```diff
@@ -85,16 +92,47 @@ type Options struct {
 	// Has no effect for single connections.
 	Concurrency uint
 
-	// Read the TLS certificate from this file
+	// Parameters used to configure TLS connections to EdgeDB server.
+	TLSOptions TLSOptions
+
+	// Read the TLS certificate from this file.
+	// DEPRECATED, use TLSOptions.CAFile instead.
 	TLSCAFile string
 
-	// If false don't verify the server's hostname when using TLS.
+	// Specifies how strict TLS validation is.
+	// DEPRECATED, use TLSOptions.Security instead.
 	TLSSecurity string
 
 	// ServerSettings is currently unused.
 	ServerSettings map[string][]byte
 }
 
+// TLSOptions contains the parameters needed to configure TLS on EdgeDB
+// server connections.
+type TLSOptions struct {
+	// PEM-encoded CA certificate
+	CA []byte
+	// Path to a PEM-encoded CA certificate file
+	CAFile string
+	// Determines how strict we are with TLS checks
+	Security TLSSecurityMode
+}
+
+// TLSSecurityMode specifies how strict TLS validation is.
+type TLSSecurityMode string
+
+const (
+	// Default makes security mode inferred from other options
+	Default TLSSecurityMode = "default"
+	// Insecure results in no certificate verification whatsoever
+	Insecure TLSSecurityMode = "insecure"
+	// NoHostVerification enables certificate verification against CAs,
+	// but hostname matching is not performed.
+	NoHostVerification TLSSecurityMode = "no_host_verification"
+	// Strict enables full certificate and hostname verification.
+	Strict TLSSecurityMode = "strict"
```

Comment:
It feels strange to me to have `edgedb.Default` instead of a more specific name `edgedb.DefaultTLSSecurity` or something. We could easily end up with other types that have a default value.

Reply:
Yeah, I forgot Golang doesn't actually namespace enums.

Reply:
Fixed.  I also uppercased the constants.